docs/repo-specific-configuration.md

```diff
@@ -32,6 +32,17 @@ You can add `<YOUR_REPO>/.scala-steward.conf` to configure how Scala Steward upd
 #pullRequests.frequency = "0 0 ? * 3" # every thursday on midnight
 pullRequests.frequency = "7 days"
 
+# Granular control of frequency based on group id and optional artifact id.
+# First-matched entry is used.
+# More-specific entry should be placed before less-specific entry.
+# `pullRequests.frequency` is used as default frequency, if none match.
+#
+# Default: empty
+pullRequests.frequencyPerGroup = [
```

Comment:
Maybe `pullRequests.perGroup` for new feature ? 🤔

Reply:
Do you mean something like this
```
pullRequests.perGroup = [
  { pattern = { groupId = "com.example", artifactId = "foo" }, config = { frequencey = "14 days", <other fields> } },
  ...
]
```
?
It looks a bit complicated but would allow to add other fields besides `frequency` that could be set per artifact.

Reply:
Yes, I mean that.
Maybe more flat 
```
{ groupId = "com.example", artifactId = "foo", frequencey = "14 days", foo = "bar", buz = 42 }
```

Reply:
One possible use case is https://github.com/scala-steward-org/scala-steward/issues/1879
```
  pattern = { groupId = "com.example", artifactId = "foo" },
  frequencey = "14 days",
  resolver = "https://repo1.maven.org/maven2/",
}
```

Reply:
`frequencyPerGroup` is a bit misleading, as it can be more specific.
And it would be great to find a better name than `pattern`

Reply:
> And it would be great to find a better name than pattern
Maybe `matcher`, or  `condition`.
I think `pattern` is not a bad choice since it corresponds to `case class UpdatePattern`.

Reply:
Renamed to more generic name `pullRequests.perGroup`.

Reply:
I think the `perGroup` change goes into the right direction since it allows to override more than one config option for dependencies that match a given pattern. How about going one step further and move the `GroupPullRequestConfig` class from `PullRequestConfig` to `RepoConfig` so that we could potentially override all config options for matching dependencies?

I imagine that it would be similar how one can override options in `.scalafmt.conf` for specific files via [`fileOverride`](https://scalameta.org/scalafmt/docs/configuration.html#fileoverride). For example, we could specify a different pull request frequency and commit message for a given dependency via a new `RepoConfig.dependencyOverride` field like this:
```
dependencyOverride = [
  {
    dependency: { groupId = ..., artifactId = ...},
    override: { pullRequests.frequency = ..., commits.message = ... }
  }, ...
]
```
The only problem I see here is that the word `override` is not appropriate
* if we introduce options that can ony be set in a `override` section
* since it implies that all options can be overridden, while it is only true for options where overriding is implemented

WDYT @exoego, @mzuehlke?

Reply:
Done in https://github.com/scala-steward-org/scala-steward/pull/2515/commits/4605e78153d6bfb11d43703afb5229b126055b81

I keep this PR for only `pullRequests` override.
`commits` support can be added in subsequent PR. 

> The only problem I see here is that the word override is not appropriate

As an user, it won't be a problem for me if it is documented that which options can be overridden

Reply:
You went with slightly different wording:
```
dependencyOverride = [
  {
    pattern = { groupId = ..., artifactId = ...},
    pullRequests.frequency = ..., 
    commits.message = ... // not there, yet
  }, ...
]
```
How about renaming `pattern` to `dependency` as in the comment from @fthomas. But for the rest keep your structure (without the `override` as `dependencyOverride` signals that we re dealing with overrides:
```
dependencyOverride = [
  {
    dependency = { groupId = ..., artifactId = ...},
    pullRequests.frequency = ..., 
    commits.message = ... // not there, yet
  }, ...
]
```
Other than this, looks good.

Reply:
Done in https://github.com/scala-steward-org/scala-steward/pull/2515/commits/cd549f040d5dc7a57ebb9acb2831271938cbca71

Reply:
This LGTM, @exoego. And sorry for the late reply.